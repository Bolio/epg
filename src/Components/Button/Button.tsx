import { MouseEventHandler } from "react";
import styled from "styled-components";

interface IProps {
  children: React.ReactNode;
  type?: "button" | "submit" | "reset";
  handleClick: MouseEventHandler<HTMLButtonElement>;
  backgroundcolor?: string;
}

const StyledButton = styled.button<{ backgroundcolor?: string }>`
  background-color: ${({ backgroundcolor }) => backgroundcolor};
  color: white;
  padding: 10px 20px;
  border: none;
  border-radius: 5px;
  cursor: pointer;
  font-size: 16px;
`;

const Button = ({ children, handleClick, backgroundcolor }: IProps) => {
  return (
    <StyledButton onClick={handleClick} backgroundcolor={backgroundcolor}>
      {children}
    </StyledButton>
  );
};
export default Button;
