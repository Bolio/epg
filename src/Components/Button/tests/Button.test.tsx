import { render, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import Button from "../Button";

describe("Button component", () => {
  it("should render the children text", () => {
    const text = "Click Me";
    const { getByText } = render(
      <Button handleClick={() => {}}>{text}</Button>
    );
    expect(getByText(text)).toBeInTheDocument();
  });

  it("Should call the handleClick function when clicked", () => {
    const handleClick = jest.fn();
    const text = "Click Me";
    const { getByText } = render(
      <Button handleClick={handleClick}>{text}</Button>
    );

    const buttonElement = getByText(text);
    fireEvent.click(buttonElement);
    expect(handleClick).toHaveBeenCalledTimes(1);
  });

  it("Should apply the background color", () => {
    const text = "Click Me";
    const backgroundcolor = "blue";
    const { getByText } = render(
      <Button handleClick={() => {}} backgroundcolor={backgroundcolor}>
        {text}
      </Button>
    );

    const buttonElement = getByText(text);
    expect(buttonElement).toHaveStyle(`background-color: ${backgroundcolor}`);
  });
});
