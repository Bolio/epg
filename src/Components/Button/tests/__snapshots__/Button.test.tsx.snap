// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Should match the snapshot 1`] = `
<DocumentFragment>
  <button
    backgroundcolor="blue"
    class="sc-beySPh iEjciT"
  >
    Click me
  </button>
</DocumentFragment>
`;
